userspace/libsinsp/logger.cpp

```diff
@@ -205,3 +271,84 @@ char* sinsp_logger::format(const char* fmt, ...)
 
 	return m_tbuf;
 }
+
+const char *sinsp_logger::encode_severity(const severity sev)
+{
+	// All severity strings should be ENCODE_LEN chars long
+	static const std::string pre("SEV");
+	static const std::string fatal_str(pre+"FAT");
+	static const std::string critical_str(pre+"CRI");
+	static const std::string error_str(pre+"ERR");
+	static const std::string warning_str(pre+"WAR");
+	static const std::string notice_str(pre+"NOT");
+	static const std::string info_str(pre+"INF");
+	static const std::string debug_str(pre+"DEB");
+	static const std::string trace_str(pre+"TRA");
+	static const std::string default_str(pre+"DEF");
+
+	const std::string *ret = nullptr;
+	switch (sev)
+	{
+	case SEV_FATAL:
+		ret = &fatal_str;
+		break;
+	case SEV_CRITICAL:
+		ret = &critical_str;
+		break;
+	case SEV_ERROR:
+		ret = &error_str;
+		break;
+	case SEV_WARNING:
+		ret = &warning_str;
+		break;
+	case SEV_NOTICE:
+		ret = &notice_str;
+		break;
+	case SEV_INFO:
+		ret = &info_str;
+		break;
+	case SEV_DEBUG:
+		ret = &debug_str;
+		break;
+	case SEV_TRACE:
+		ret = &trace_str;
+		break;
+	default:
+		ASSERT(false);
+		ret = &default_str;
+		break;
+	}
+
+	// Ideally this would be a compile time assert,
+	// but that's not possible with std::string
+	if (ret->length() != ENCODE_LEN)
+	{
+		throw sinsp_exception("Invalid encoding string, len "
+				      + std::to_string(ret->length()));
+	}
+	return ret->c_str();
+}
+
+sinsp_logger::severity sinsp_logger::decode_severity(const std::string &str, size_t *len)
```

Comment:
Since you're returning values that are outside of the `sinsp_logger::severity` type, you might consider just returning `int`.  Client code might not think to consider a case where a value of type `sinsp_logger::severity` is `SEV_MAX + 1`.